compiler/luci/pass/src/SubstituteExpandDimsToReshapePass.cpp

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/SubstituteExpandDimsToReshapePass.h"
+
+#include <luci/IR/CircleNodes.h>
+#include <luci/Profile/CircleNodeOrigin.h>
+
+/**
+ * @brief Convert expand_dims op to reshape op
+ *        (expand_dims op with const axis ONLY)
+ * @example
+ *             input.shape = [2,3,4]
+ *             expand_dims(input, axis=1)
+ *
+ *             can be converted to
+ *
+ *             reshape(input, [2,1,3,4])
+ */
+namespace
+{
+
+int32_t unknown_dim_count(luci::CircleNode *node)
+{
+  int32_t count = 0;
+
+  for (uint32_t i = 0; i < node->rank(); ++i)
+    if (!node->dim(i).known())
+      ++count;
+
+  return count;
+}
+
+/**
+ * @brief Return value in position on CircleConst with int32 format.
+ */
+int32_t value_from_circle_const(const luci::CircleConst *node, uint32_t idx)
+{
+  assert(node->rank() == 1 && node->dim(0).value() > idx);
+  assert(node->dtype() == loco::DataType::S64 || node->dtype() == loco::DataType::S32);
+
+  if (node->dtype() == loco::DataType::S64)
+    return node->at<loco::DataType::S64>(idx);
+  return static_cast<int32_t>(node->at<loco::DataType::S32>(idx));
+}
+
+bool substitute_expand_dims_to_reshape(luci::CircleNode *node)
+{
+  auto target_node = dynamic_cast<luci::CircleExpandDims *>(node);
+  if (target_node == nullptr)
+    return false;
+  auto input_node = loco::must_cast<luci::CircleNode *>(target_node->input());
+  if (input_node->rank() <= 0)
+    return false;
+  auto axis_node = dynamic_cast<luci::CircleConst *>(target_node->axis());
+  if (axis_node == nullptr)
+    return false;
+
+  int32_t axis = value_from_circle_const(axis_node, 0);
+  if (axis < 0)
+    axis = axis + static_cast<int32_t>(input_node->rank()) + 1;
```

Comment:
why +1 ?

Reply:
Example) 
input node: {1, 3}
input node's rank: 2
axis: -1

axis should point to the last index of **new** input node.
expected new input node: {1, 3, 1}
expected last index: 2

Likewise, axis must be the current last index + 1. Therefore (input_node->rank() + 1).

Reply:
as `+1` is not like other Ops axis with -1, it would be nice to have a comment explaining this in the source.